### PR TITLE
New Timezonetime IA

### PR DESCRIPTION
**Pull request by pjhampton**
## Description of new Instant Answer, or changes

Finish's the Timezone/time IA.

## People to notify

@moollaza 

## Testing & Review
To be completed by Community Leader (or DDG Staff) when reviewing Pull Request

**Pull Request**
- [x] Title follows correct format (Specifies Instant Answer + Purpose)
- [x] Description contains a valid Instant Answer Page Link (e.g. https://duck.co/ia/view/my_ia)

**Instant Answer Page** (for new Instant Answers)
- [x] Instant Answer page is correctly filled out and contains:
    - The **Title** is appropriately named and formatted
    - The **IA topics** are present and appropriate
    - The **Description** is clear and coherent
    - **Source Name** exists if applicable
    - All **Example Queries** trigger on [Beta](https://beta.duckduckgo.com/)

**Code**
- [ ] Adheres to the [DuckDuckGo Style Guide](https://docs.duckduckhack.com/resources/code-style-guide.html)
- [ ] Behaviour is appropriately tested. If improvement, tests are adequately extended.
- [ ] There is no unnecessary files in place (such as editor config files)
- [ ] There is no API keys / secrets present
- [ ] Tests are passing (run `$ duckpan test <goodie_id>`)
    - Tester should report any failures

**Ready to merge?**

- [ ] Has this IA been deployed to and tested on [beta.duckduckgo.com](https://beta.duckduckgo.com/)?
- [ ] For larger commits, has this been approved be more than one community member?
- [ ] The number of reviews is appropriate for this type of PR
- [ ] The commit message is clear, coherent and fitting

**Pull Request Review Guidelines**: https://docs.duckduckhack.com/programming-mission/pr-review.html

---

Instant Answer Page: https://duck.co/ia/view/timezonetime

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: lib/DDG/Goodie/Timezonetime.pm [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    "GMT" => "GMT",
+    "BST" => "Europe/London", 
+    "PST" => "PST8PDT", 
+    "CST" => "CST6CDT"
+};
+
+handle query_lc => sub {
+    my ($query) = $_;
+    
+    my $timezones = join('|', keys(%$timezoneMapping));
```
Comment (moollaza) [inline lib/DDG/Goodie/Timezonetime.pm]: This should be done outside the `handle` so it's cached in memory 👍

File: lib/DDG/Goodie/Timezonetime.pm [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    "IST" => "Asia/Kolkata", 
+    "EST" => "EST", 
+    "UTC" => "UTC", 
+    "GMT" => "GMT",
+    "BST" => "Europe/London", 
+    "PST" => "PST8PDT", 
+    "CST" => "CST6CDT"
+};
+
+handle query_lc => sub {
```
Comment (moollaza) [inline lib/DDG/Goodie/Timezonetime.pm]: I think we should handle `remainder`? then we don't need to do the replacement on ln 35.

Assume the remainder is the timezone and try and use it to try a lookup in the hash.

File: lib/DDG/Goodie/Timezonetime.pm [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    my ($query) = $_;
+    
+    my $timezones = join('|', keys(%$timezoneMapping));
+    my $daylightStatus = "";
+    
+    # Parsing timezone out of query string
+    $query =~ s/.*\b($timezones)\b.*/$1/ig;
+    my $timezone = uc($query);
```
Comment (moollaza) [inline lib/DDG/Goodie/Timezonetime.pm]:
```perl
my $timezone = $timezoneMapping->{uc $_} // 0;
return unless $timezone;
```

**Comment by moollaza** [comment]
Live on Beta: https://beta.duckduckgo.com/?q=time+in+ist&ia=

**Comment by moollaza** [comment]
@pjhampton Thanks! This LGTM :+1:

**Comment by bsstoner** [comment]
This was released, do we need to do something with IA pages to enable it?

**Comment by pjhampton** [comment]
@bsstoner cool 😄  I have moved this to online in the duck.co/ia pages.

**Pull request merged**